src/components/PipelineRun/PipelineRunListView/PipelineRunsListView.tsx

```diff
@@ -37,6 +38,52 @@ type PipelineRunsListViewProps = {
   customFilter?: (plr: PipelineRunKind) => boolean;
 };
 
+const getPipelineRunSortFunction = (
+  columnKey: PipelineRunColumnKeys,
+  direction: SortByDirection,
+) => {
+  const getValue = (plr: PipelineRunKind) => {
+    switch (columnKey) {
+      case 'name':
+        return plr.metadata?.name || '';
+      case 'started':
+        return plr.status?.startTime || '';
+      case 'duration':
+        return plr.status?.completionTime && plr.status?.startTime
+          ? new Date(plr.status.completionTime).getTime() - new Date(plr.status.startTime).getTime()
+          : 0;
+      case 'status':
+        return pipelineRunStatus(plr) || '';
+      case 'type':
+        return plr.metadata?.labels?.[PipelineRunLabel.PIPELINE_TYPE] || '';
+      case 'component':
+        return plr.metadata?.labels?.[PipelineRunLabel.COMPONENT] || '';
+      case 'trigger':
+        return plr.metadata?.labels?.[PipelineRunLabel.COMMIT_EVENT_TYPE_LABEL] || '';
+      case 'reference':
+        return (
+          plr.metadata?.labels?.[PipelineRunLabel.COMMIT_REPO_URL_LABEL] ||
+          plr.metadata?.annotations?.[PipelineRunLabel.COMMIT_FULL_REPO_URL_ANNOTATION] ||
+          ''
+        );
+      default:
+        return '';
+    }
+  };
+
+  return (a: PipelineRunKind, b: PipelineRunKind) => {
+    const aValue = getValue(a);
+    const bValue = getValue(b);
+
+    if (typeof aValue === 'number' && typeof bValue === 'number') {
+      return direction === SortByDirection.asc ? aValue - bValue : bValue - aValue;
+    }
+
+    const comparison = String(aValue).localeCompare(String(bValue));
```

Comment:
same point I mentioned previously: this might not work with date comparison (`started`, `duration`)